Updated auto_auth endpoint to always return JSON

clintonb opened the pull request:
Defaulting to a plaintext response makes no sense for an endpoint that is intended to be used by machines for testing. The endpoint now returns JSON only unless a redirect action is triggered.

clintonb commented:
@pwnage101 @jibsheet please review
FYI @edx/testeng

clintonb commented:
jenkins run a11y

clintonb commented:
jenkins run bokchoy

jibsheet commented:
Looks like the a11y/bokchoy might actually be change related?  
```Timed out waiting to load page '<common.test.acceptance.pages.lms.auto_auth.AutoAuthPage```

clintonb commented:
Yep. I just noticed that. Fixing it now.

pwnage101 commented:
are you running isort or something?

clintonb commented:
Yes. See the first commit.

clintonb commented:
Import optimization moved to https://github.com/edx/edx-platform/pull/15203.

clintonb commented:
@pwnage101 @jibsheet I _finally_ fixed the issues with the four failing dashboard tests. The use of fixtures hid the fact that `EcommerceTest` requires the user to have the `finance_admin` `CourseAccessRole`. To avoid future confusion I have deleted the fixture and updated the tests to create a user with the correct roles.

Issues like this are precisely why we should do away with fixtures or, at a minimum, document the fact that tests rely on fixtures in the test classes/methods. Fixtures make tests brittle and refactoring difficult.
